### PR TITLE
Invoice Viewing complete

### PR DESCRIPTION
**Pull request by Austish101**
#31 #32
Can now view invoices/operations with filters of all/nhs/private, and start/end dates to search between.

### CODE REVIEW

File: esd-group-3/src/java/models/Operation.java [hunk 1]
```diff
@@ -164,15 +174,99 @@ public void retrieveByOperationId(DBConnection dbcon, int opId) {
...
+//            }
+//            
+////            int row_count = countAllOperations(dbcon);
+////            for (int i = 0; i < row_count; i++) {
+////                Operation tempOp = new Operation();
+////                tempOp.retrieveByOperationId(dbcon, (i+1));
+////                operationsArray.add(tempOp);
+////            }
+//        }
+//        else {
```
Comment (conranpearce) [inline esd-group-3/src/java/models/Operation.java]: Need to remove comments

File: esd-group-3/src/java/com/IssueInvoiceServlet.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          
+        try {
+            DBConnection dbcon = new DBConnection("smartcaretest", "", "");
+            Operation new_op = new Operation();
+            HttpSession loginSession = request.getSession();
+            String employee_name = (String) loginSession.getAttribute("name");
+            
+            //  dbcon, get from employee_name above, html select, select, select, select, work out charge, select slot, init paid as false?
+            // new_op.create(dbcon, employee_userid, client_userid, date, starttime, endtime, 0, 0, false);
+
```
Comment (conranpearce) [inline esd-group-3/src/java/com/IssueInvoiceServlet.java]: Need to remove these comments?

File: esd-group-3/web/addInvoice.jsp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                   min="2018-01-01" max="2021-12-31">
+            <label for="start">End date:</label>
+            <input type="date" id="end" name="end"
+                   value="2018-07-22"
+                   min="2018-01-01" max="2021-12-31">
+            
+            <form action="InvoiceViewerServlet" method="GET">
+                <input type="submit" value="Update" class="button logout"> 
+            </form>
+        </div>-->
```
Comment (conranpearce) [inline esd-group-3/web/addInvoice.jsp]: Need to remove the comments

**Review by conranpearce — CHANGES_REQUESTED** [review]

File: esd-group-3/src/java/com/IssueInvoiceServlet.java (hunk 2 above)
Comment (Austish101) [reply esd-group-3/src/java/com/IssueInvoiceServlet.java]: This file isn't finished yet as it's going to be part of another ticket too, so these comments are just here for future me/someone to do.

File: esd-group-3/web/addInvoice.jsp (hunk 3 above)
Comment (Austish101) [reply esd-group-3/web/addInvoice.jsp]: This file isn't finished yet as it's going to be part of another ticket too, so these comments are just here for future me/someone to do.

File: esd-group-3/src/java/models/Operation.java (hunk 1 above)
Comment (Austish101) [reply esd-group-3/src/java/models/Operation.java]: Removed.

**Review by conranpearce — APPROVED** [review]

**Pull request merged**